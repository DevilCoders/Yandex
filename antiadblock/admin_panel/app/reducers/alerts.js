import update from 'app/lib/update';

import * as alertsActions from 'app/actions/alerts';

export default function(state, action) {
    const {type} = action;

    state = state || {
        items: {},
        loaded: false
    };

    switch (type) {
        case alertsActions.START_ALERTS_LOADING: {
            return state;
        }
        case alertsActions.END_ALERTS_LOADING: {
            if (!action || typeof action.items !== 'object') {
                return state;
            }

            return update(state, {
               $merge: {
                    items: action.items,
                    loaded: true,
                    lastUpdateTimeAlerts: new Date()
               }
            });
        }
        default: {
            return state;
        }
    }
}
