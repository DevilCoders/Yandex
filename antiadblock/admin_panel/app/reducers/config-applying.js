import * as configApplyingActions from 'app/actions/config-applying';
import update from 'app/lib/update';

export default function(state, action) {
    const {type} = action;
    const initialState = {
        visible: false,
        id: null,
        serviceId: null,
        target: null,
        configData: null,
        oldConfigData: null,
        loaded: false
    };

    state = state || initialState;

    switch (type) {
        case configApplyingActions.OPEN_CONFIG_APPLYING: {
            return {
                visible: true,
                id: action.id,
                serviceId: action.serviceId,
                target: action.target,
                configData: action.configData,
                oldConfigData: null,
                loaded: false
            };
        }
        case configApplyingActions.CLOSE_CONFIG_APPLYING: {
            return initialState;
        }
        case configApplyingActions.END_ACTIVE_CONFIG_LOADING: {
            return update(state, {
                $merge: {
                    oldConfigData: action.config,
                    loaded: true
                }
            });
        }
        default: {
            return state;
        }
    }
}
