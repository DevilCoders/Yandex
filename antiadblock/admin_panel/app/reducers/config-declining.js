import * as configDecliningActions from 'app/actions/config-declining';

export default function(state, action) {
    const {type} = action;
    const initialState = {
        visible: false,
        id: null,
        serviceId: null,
        labelId: null,
        approved: null
    };

    state = state || initialState;

    switch (type) {
        case configDecliningActions.OPEN_CONFIG_DECLINING: {
            return {
                visible: true,
                id: action.id,
                labelId: action.labelId,
                serviceId: action.serviceId,
                approved: action.approved
            };
        }
        case configDecliningActions.CLOSE_CONFIG_DECLINING: {
            return initialState;
        }
        default: {
            return state;
        }
    }
}
