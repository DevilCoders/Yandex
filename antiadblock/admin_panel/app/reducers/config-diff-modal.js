import * as configDiffModalActions from 'app/actions/config-diff-modal';

export default function(state, action) {
    const {type} = action;
    const initialState = {
        id: null,
        secondId: null,
        serviceId: null,
        modalVisible: false
    };

    state = state || initialState;

    switch (type) {
        case configDiffModalActions.OPEN_CONFIG_DIFF_MODAL: {
            return {
                id: action.id,
                secondId: action.secondId,
                serviceId: action.serviceId,
                modalVisible: true
            };
        }
        case configDiffModalActions.CLOSE_CONFIG_DIFF_MODAL: {
            return initialState;
        }
        default: {
            return state;
        }
    }
}
