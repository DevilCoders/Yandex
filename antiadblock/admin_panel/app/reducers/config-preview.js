import * as configPreviewActions from 'app/actions/config-preview';

export default function(state, action) {
    const {type} = action;
    const initialState = {
        visible: false,
        id: null,
        serviceId: null,
        configData: null
    };

    state = state || initialState;

    switch (type) {
        case configPreviewActions.OPEN_CONFIG_PREVIEW: {
            return {
                visible: true,
                id: action.id,
                serviceId: action.serviceId,
                configData: action.configData
            };
        }
        case configPreviewActions.CLOSE_CONFIG_PREVIEW: {
            return initialState;
        }
        default: {
            return state;
        }
    }
}
