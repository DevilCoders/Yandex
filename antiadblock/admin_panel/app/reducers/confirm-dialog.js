import * as confirmDialogActions from 'app/actions/confirm-dialog';

export default function(state, action) {
    const {type} = action;
    const initialState = {
        visible: false,
        message: null,
        callback: null
    };

    state = state || initialState;

    switch (type) {
        case confirmDialogActions.OPEN_CONFIRM_DIALOG: {
            return {
                visible: true,
                message: action.message,
                callback: action.callback
            };
        }
        case confirmDialogActions.CLOSE_CONFIRM_DIALOG: {
            return initialState;
        }
        default: {
            return state;
        }
    }
}
