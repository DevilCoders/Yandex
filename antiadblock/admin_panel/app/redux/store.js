import {createStore, compose, applyMiddleware} from 'redux';
import reducer from 'app/reducers/index';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';

let composeEnhancers = compose;

export function configureStore(state = {}) {
    if (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION__) {
        composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__;
    }
    if (process.env.NODE_ENV === 'local') {
        return createStore(reducer, state, composeEnhancers(applyMiddleware(thunkMiddleware, createLogger())));
    }
    return createStore(reducer, state, composeEnhancers(applyMiddleware(thunkMiddleware)));
}
