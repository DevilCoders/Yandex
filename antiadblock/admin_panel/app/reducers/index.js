import {combineReducers} from 'redux';

import alerts from './alerts';
import service from './service';
import services from './services';
import configPreview from './config-preview';
import configApplying from './config-applying';
import configDiffModal from './config-diff-modal';
import ticketCreation from './ticket-creation';
import confirmDialog from './confirm-dialog';
import configDeclining from './config-declining';
import createLabel from './create-label';
import experimentApplying from './experiment-applying';
import errors from './errors';
import metric from './metric';
import search from './search';
import heatmap from './heatmap';
import user from './user';
import profileDiffModal from './profile-diff-modal';

import config from './config/config';

const appReducer = combineReducers({
    service,
    services,
    configPreview,
    configApplying,
    configDiffModal,
    confirmDialog,
    configDeclining,
    createLabel,
    experimentApplying,
    metric,
    errors,
    search,
    heatmap,
    alerts,
    user,
    config,
    profileDiffModal,
    ticketCreation
});

export default appReducer;

export function getService(state) {
    return state.service;
}

export function getServices(state) {
    return state.services;
}

export function getConfigPreview(state) {
    return state.configPreview;
}

export function getConfigApplying(state) {
    return state.configApplying;
}

export function getConfigDiffModal(state) {
    return state.configDiffModal;
}

export function getConfigDeclining(state) {
    return state.configDeclining;
}

export function getExperimentApplying(state) {
    return state.experimentApplying;
}

export function getTicketCreation(state) {
    return state.ticketCreation;
}

export function getMetric(state, serviceId) {
    return state.metric[serviceId];
}

export function getErrors(state) {
    return state.errors;
}

export function getSearch(state) {
    return state.search;
}

export function getHeatmap(state) {
    return state.heatmap;
}

export function getSearchById(state, id) {
    return state.search.byId[id];
}

export function getAlerts(state) {
    return state.alerts;
}

export function getUser(state) {
    return state.user;
}

export function getConfig(state) {
    return state.config;
}

export function getProfileDiffModal(state) {
    return state.profileDiffModal;
}

export function getCreateLabel(state) {
    return state.createLabel;
}
