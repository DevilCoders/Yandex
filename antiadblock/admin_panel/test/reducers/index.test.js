import {getService, getServices, getConfigPreview, getConfigDiffModal, getConfigApplying, getMetric, getErrors} from 'app/reducers/index';

import {ROOT_STATE} from '../fixtures/store';

describe('index', () => {
    const initialState = ROOT_STATE;

    // TODO test combine reducers

    describe('selectors', () => {
        test('getService should return service obj', () => {
            expect(getService(initialState)).toMatchSnapshot();
        });

        test('getServices should return services obj', () => {
            expect(getServices(initialState)).toMatchSnapshot();
        });

        test('getConfigPreview should return configPreview obj', () => {
            expect(getConfigPreview(initialState)).toMatchSnapshot();
        });

        test('getConfigApplying should return configApplying obj', () => {
            expect(getConfigApplying(initialState)).toMatchSnapshot();
        });

        test('getConfigDiffModal should return configDiffModal obj', () => {
            expect(getConfigDiffModal(initialState)).toMatchSnapshot();
        });

        test('getMetric should return metric obj', () => {
            expect(getMetric(initialState, 'test.service')).toMatchSnapshot();
        });

        test('getErrors should return errors obj', () => {
            expect(getErrors(initialState)).toMatchSnapshot();
        });
    });
});
