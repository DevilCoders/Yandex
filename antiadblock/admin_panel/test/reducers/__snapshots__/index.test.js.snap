// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index selectors getConfigApplying should return configApplying obj 1`] = `
Object {
  "activeConfigData": null,
  "configData": null,
  "id": null,
  "loaded": false,
  "serviceId": null,
  "visible": false,
}
`;

exports[`index selectors getConfigDiffModal should return configDiffModal obj 1`] = `
Object {
  "configsData": Object {},
  "id": null,
  "loaded": false,
  "oldId": null,
  "serviceId": null,
  "visible": false,
}
`;

exports[`index selectors getConfigPreview should return configPreview obj 1`] = `
Object {
  "configData": null,
  "id": null,
  "serviceId": null,
  "visible": false,
}
`;

exports[`index selectors getErrors should return errors obj 1`] = `
Object {
  "byId": Object {
    "global": Array [
      "global message",
    ],
    "id1": Array [
      "message 1",
      "message 2",
    ],
    "id2": Array [
      "message 3",
      "message 4",
    ],
  },
}
`;

exports[`index selectors getMetric should return metric obj 1`] = `
Object {
  "http_codes": Object {
    "data": Object {
      "1524759180000": Object {
        "200": 2368,
        "204": 85,
        "301": 182,
        "302": 11,
        "304": 1,
        "404": 173,
      },
      "1524759240000": Object {
        "200": 2797,
        "204": 100,
        "301": 232,
        "302": 20,
        "304": 7,
        "404": 206,
      },
    },
    "loaded": true,
    "loading": false,
    "range": 10,
  },
}
`;

exports[`index selectors getService should return service obj 1`] = `
Object {
  "applyingConfig": Object {
    "id": null,
    "progress": false,
    "target": null,
  },
  "comment": "",
  "configs": Object {
    "byId": Object {},
    "items": Array [],
    "loaded": false,
    "offset": 0,
    "total": 0,
  },
  "loaded": false,
  "service": null,
  "setup": Object {
    "loaded": false,
  },
}
`;

exports[`index selectors getServices should return services obj 1`] = `
Object {
  "items": Array [],
  "loaded": false,
  "loading": false,
}
`;
